File: 1436/solution.go
package main

func destCity(paths [][]string) string {
	dests := make(map[string]bool)
	for _, path := range paths {
		dests[path[1]] = true
		dests[path[0]] = true
	}
	for _, path := range paths {
		dests[path[0]] = false
	}

	for key, val := range dests {
		if val {
			return key
		}
	}
	return "None"
}

File: 1436/solution.ts
function destCity(paths: string[][]): string {
  const cities = new Set<string>();

  // Collect outgoing cities
  for (const path of paths) {
    cities.add(path[0]);
  }

  // Find destination city with no outgoing path
  for (const path of paths) {
    const dest = path[1];
    if (!cities.has(dest)) {
      return dest;
    }
  }

  return "";
}
